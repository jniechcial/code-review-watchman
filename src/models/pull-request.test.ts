import PullRequest from './pull-request';
import Review from './review';

const stubbedData = {
    id: '123',
    url: 'testURL',
    reviews: [
        { user: { login: 'test1' }, state: 'CHANGES_REQUESTED' },
        { user: { login: 'test1' }, state: 'APPROVED' },
        { user: { login: 'test2' }, state: 'CHANGES_REQUESTED' },
        { user: { login: 'test2' }, state: 'SOME_OTHER_STATE' },
    ],
};

describe('PullRequest', () => {
    it('creates pull request instance with reviews', async () => {
        const pullRequest = new PullRequest(stubbedData.id, stubbedData.url, stubbedData.reviews);
        expect(pullRequest.reviews[0]).toBeInstanceOf(Review);
        expect(pullRequest.reviews).toMatchSnapshot();
    });

    it('#rejectedReviews returns all rejected reviews', async () => {
        const pullRequest = new PullRequest(stubbedData.id, stubbedData.url, stubbedData.reviews);
        expect(pullRequest.rejectedReviews()).toMatchSnapshot();
    });

    it('#approvedReviews returns all approved reviews', async () => {
        const pullRequest = new PullRequest(stubbedData.id, stubbedData.url, stubbedData.reviews);
        expect(pullRequest.approvedReviews()).toMatchSnapshot();
    });

    it('#getReviewsByUsername returns all reviews of particular user', async () => {
        const pullRequest = new PullRequest(stubbedData.id, stubbedData.url, stubbedData.reviews);
        expect(pullRequest.getReviewsByUsername('test1').length).toMatchSnapshot();
    });
});
