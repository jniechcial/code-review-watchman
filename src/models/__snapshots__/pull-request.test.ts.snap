// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequest #approvedReviews returns all approved reviews 1`] = `
Array [
  PullRequestReview {
    "state": "APPROVED",
    "username": "test1",
  },
]
`;

exports[`PullRequest #getReviewsByUsername returns all reviews of particular user 1`] = `2`;

exports[`PullRequest #rejectedReviews returns all rejected reviews 1`] = `
Array [
  PullRequestReview {
    "state": "CHANGES_REQUESTED",
    "username": "test1",
  },
  PullRequestReview {
    "state": "CHANGES_REQUESTED",
    "username": "test2",
  },
]
`;

exports[`PullRequest creates pull request instance with reviews 1`] = `
Array [
  PullRequestReview {
    "state": "CHANGES_REQUESTED",
    "username": "test1",
  },
  PullRequestReview {
    "state": "APPROVED",
    "username": "test1",
  },
  PullRequestReview {
    "state": "CHANGES_REQUESTED",
    "username": "test2",
  },
  PullRequestReview {
    "state": "SOME_OTHER_STATE",
    "username": "test2",
  },
]
`;
